cardano-cli/src/Cardano/CLI/Shelley/Run/Query.hs

```diff
@@ -208,26 +210,35 @@ runQueryTip
 runQueryTip (AnyConsensusModeParams cModeParams) network mOutFile = do
   SocketPath sockPath <- firstExceptT ShelleyQueryCmdEnvVarSocketErr readEnvSocketPath
   let localNodeConnInfo = LocalNodeConnectInfo cModeParams network sockPath
-      consensusMode = consensusModeOnly cModeParams
-
-  anyEra <- determineEra cModeParams localNodeConnInfo
-  tip <- liftIO $ getLocalChainTip localNodeConnInfo
-
-  let tipSlotNo = case tip of
-        ChainTipAtGenesis -> 0
-        ChainTip slotNo _ _ -> slotNo
+  let consensusMode = consensusModeOnly cModeParams
 
   case consensusMode of
     CardanoMode -> do
-        eResult <- liftIO $ queryEraHistoryAndSystemStart localNodeConnInfo Nothing
-        case eResult of
-          Left acqFail -> left (ShelleyQueryCmdAcquireFailure acqFail)
-          Right (eraHistory, mSystemStart) ->
+        (chainTip, result) <- liftIO $ queryChainTipAndEraHistoryAndSystemStart localNodeConnInfo Nothing
+
+        let tipSlotNo = case chainTip of
+              ChainTipAtGenesis -> 0
+              ChainTip slotNo _ _ -> slotNo
+
+        case join (either (const Nothing) Just <$> result) of
```

Comment:
What we should do to avoid the gymnastics here is use `ShelleyQueryCmdError` ( i.e `ShelleyQueryCmdAcquireFailure`) instead of the `Maybe` in `singleQuery`.  This will help simplify the code and we can deal with the case (more clearly imo) where the node to client version < 9.

Reply:
These cases are not failures though.  Does `ShelleyQueryCmdError` still make sense?